### PR TITLE
Refactor Vulkan validation layer loading

### PR DESCRIPTION
**Pull request by akioCL**
## What does this PR do?

-Add support to load extensions from layers. This allows us to use the EXT_debug_utils on Android (instead of the deprecated EXT_debug_report extension), since it's not provided by the driver, but instead by the Khronos Validation layer.
-Remove support for deprecated debug extension EXT_debug_report 
-Removed deprecated layers for Android
-Update GLAD to Vulkan 1.3
-Moved FunctionLoader from Vulkan/RHI.Loader to Vulkan/RHI
-Add Ebus for collecting Vulkan instance and device requirements
-Add notification Ebus for instance creation/destruction
-Add LoaderContext class to encapsulate the loading of Vulkan function pointers

OpenXR changes on this PR https://github.com/o3de/o3de-extras/pull/422

## How was this PR tested?

Run Windows and Android using Vulkan with validation on and off.
Also run ASV Features/OpenXR sample on PC and Android

### CODE REVIEW

File: Gems/Atom/RHI/Vulkan/Code/Source/RHI/Device.cpp [hunk 1]
```diff
@@ -328,8 +328,7 @@ namespace AZ
                 AssertSuccess(vkResult);
                 RETURN_RESULT_IF_UNSUCCESSFUL(ConvertResult(vkResult));
 
-                if (!instance.GetFunctionLoader().LoadProcAddresses(
-                        &m_context, instance.GetNativeInstance(), physicalDevice.GetNativePhysicalDevice(), m_nativeDevice))
+                if (!FunctionLoader::LoadProcAddresses(&m_context, physicalDevice.GetNativePhysicalDevice(), m_nativeDevice))
```
Comment (moudgils) [inline Gems/Atom/RHI/Vulkan/Code/Source/RHI/Device.cpp:331]: Did you check to see if this didnt break VR?

File: Gems/Atom/RHI/Vulkan/Code/Source/RHI/Vulkan.cpp [hunk 2]
```diff
@@ -220,63 +220,7 @@ namespace AZ
...
-
-                /*
-                    The return value of this callback controls whether the Vulkan call that caused the validation message
-                    will be aborted or not. We return VK_FALSE as we DON'T want Vulkan calls that cause a validation message
-                    (and return a VkResult) to abort. If you instead want to have calls abort, pass in VK_TRUE and the function will
-                    return VK_ERROR_VALIDATION_FAILED_EXT.
-                */
-                return VK_FALSE;
-            }
+            const char* s_debugMessageLabel = "vkDebugMessage";
```
Comment (moudgils) [inline Gems/Atom/RHI/Vulkan/Code/Source/RHI/Vulkan.cpp:223]: How do we report without the callback?

File: Gems/Atom/RHI/Vulkan/Code/Source/RHI/FunctionLoader.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                        }
+                    }
+                }
+            }
+        }
+
+        bool FunctionLoader::LoadProcAddresses(GladVulkanContext* context, VkPhysicalDevice physicalDevice, VkDevice device)
+        {
+            VkInstance instance = Instance::GetInstance().GetNativeInstance();
+            int result = gladLoaderLoadVulkanContext(context, instance, physicalDevice, device);
```
Comment (VickyAtAZ) [inline Gems/Atom/RHI/Vulkan/Code/Source/RHI/FunctionLoader.cpp:98]: I actually prefer the previous function declaration which have VkInstance as input parameter. 
It's clear that people notice the different of the three LoadProcAddresses calls and won't wonder why there are two LoadProcAddresses calls with same parameters.

**Comment by VickyAtAZ** [comment]
> Add support to load extensions from layers. This allows us to use the EXT_debug_report on Android (instead of the deprecated EXT_debug_report extension), since it's not provided by the driver, but instead by the Khronos Validation layer.

Do you mean "This allows us to use the EXT_debug_utils on Android (instead of the deprecated EXT_debug_report extension),"

File: Gems/Atom/RHI/Vulkan/Code/Source/RHI/FunctionLoader.cpp [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+#include <RHI/Vulkan.h>
+#include <RHI/Instance.h>
+#include <RHI/FunctionLoader.h>
+
+#include <AzCore/StringFunc/StringFunc.h>
+
+namespace AZ
+{
+    namespace Vulkan
```
Comment (moudgils) [inline Gems/Atom/RHI/Vulkan/Code/Source/RHI/FunctionLoader.cpp:17]: We have LoaderContext class. Who is using FunctionLoader?

File: Gems/Atom/RPI/Code/Source/RPI.Public/RPISystem.cpp [hunk 5]
```diff
@@ -80,21 +80,6 @@ namespace AZ
 
         void RPISystem::Initialize(const RPISystemDescriptor& rpiSystemDescriptor)
         {
```
Comment (moudgils) [inline Gems/Atom/RPI/Code/Source/RPI.Public/RPISystem.cpp:82]: Make sure that if link mode is not initiated and someone passed in -openxr=enable it doesn't crash and just gracefully handles disabling xrsystem with an AZ_error message.

File: Gems/Atom/RHI/Vulkan/Code/Source/RHI/SystemComponent.cpp [hunk 6]
```diff
@@ -67,6 +67,11 @@ namespace AZ
             required.push_back(RHI::Factory::GetManagerComponentService());
         }
 
+        void SystemComponent::GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+        {
+            required.push_back(AZ_CRC_CE("VulkanRequirementsService"));
```
Comment (moudgils) [inline Gems/Atom/RHI/Vulkan/Code/Source/RHI/SystemComponent.cpp:72]: who is providing this service?

**Review by moudgils — COMMENTED** [review]
New changes look good. Were you able to test quest 2 with link mode as well as via native app?

**Review by moudgils — APPROVED** [review]

**Review by VickyAtAZ — APPROVED** [review]

**Comment by amzn-changml** [comment]
Discussed with @akioCL and @moudgils. Since this PR is implemented by a supplemental PR for O3DE-Extras, we're going to need to merge this PR and https://github.com/o3de/o3de-extras/pull/422 simultaneously. This Jenkins job [jenkins.build.o3de.org/blue/organizations/jenkins/akioCL-O3DE-PR16317-O3DE-Extras-PR422/detail/refactor_openxr_init/15/pipeline](https://jenkins.build.o3de.org/blue/organizations/jenkins/akioCL-O3DE-PR16317-O3DE-Extras-PR422/detail/refactor_openxr_init/15/pipeline) will act as the validation that it passes AR

**Pull request merged**